Source/Core/Common/ArmCPUDetect.cpp

```diff
@@ -277,6 +282,33 @@ void CPUInfo::Detect()
   {
     cpu_id = MIDRToString(midr);
   }
+#elif defined(__OpenBSD__)
+  // OpenBSD
+  int mib[2];
+  size_t len;
+  char hwmodel[256];
+  uint64_t isar0;
+
+  mib[0] = CTL_HW;
+  mib[1] = HW_MODEL;
+  len = std::size(hwmodel);
+  if (sysctl(mib, 2, &hwmodel, &len, nullptr, 0) != -1)
+    model_name = std::string(hwmodel, len - 1);
+
+  mib[0] = CTL_MACHDEP;
+  mib[1] = CPU_ID_AA64ISAR0;
+  len = sizeof(isar0);
+  if (sysctl(mib, 2, &isar0, &len, nullptr, 0) != -1)
+  {
+    if (ID_AA64ISAR0_AES(isar0) >= ID_AA64ISAR0_AES_PMULL)
+      bAES = true;
+    if (ID_AA64ISAR0_AES(isar0) >= ID_AA64ISAR0_SHA1_BASE)
+      bSHA1 = true;
+    if (ID_AA64ISAR0_AES(isar0) >= ID_AA64ISAR0_SHA2_BASE)
+      bSHA2 = true;
+    if (ID_AA64ISAR0_AES(isar0) >= ID_AA64ISAR0_CRC32_BASE)
+      bCRC32 = true;
```

Comment:
Shouldn't this be:

```suggestion
    if (ID_AA64ISAR0_AES(isar0) >= ID_AA64ISAR0_AES_BASE)
      bAES = true;
    if (ID_AA64ISAR0_SHA1(isar0) >= ID_AA64ISAR0_SHA1_BASE)
      bSHA1 = true;
    if (ID_AA64ISAR0_SHA2(isar0) >= ID_AA64ISAR0_SHA2_BASE)
      bSHA2 = true;
    if (ID_AA64ISAR0_CRC32(isar0) >= ID_AA64ISAR0_CRC32_BASE)
      bCRC32 = true;
```

Reply:
My bad. A bit too much copy and paste. I meant to correct the left side when I had initially adjusted the right side as appropriate. and with regard to AES_BASE vs AES_PMULL, that looks to be more appropriate for testing for just AES and not PMULL.